### PR TITLE
[ui] Refactor SplitPanelContainer

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

I set out to add a minimum size for the second panel in a `SplitPanelContainer`, and I fell into a refactoring rabbit hole.

- Add a minimum size (width/height) for the second pane, as originally planned.
- Rewrite into functional components, with an imperative handle on `SplitPanelContainer` to replicate its imperative API.
- Set a minimum for a few different panes around the app (global asset graph, run log pane)
- Fix up the expand/collapse panel on launchpad config, which has some tricky ref usage.
- Tidy up a few other things that I found hard to follow or confusing.

Note that this may lead to some panel size allocations being slightly different from before, in cases where we have a minimum size for the second panel. For instance, if I set a minimum of 400px for the second panel, and the first panel is set to occupy 75% of the container, that 75% will now apply to the available size *minus* 400px, instead of 75% of the full container. FWIW, I don't think this will be a noticeable change for most people.

https://github.com/dagster-io/dagster/assets/2823852/b4bbc92d-c57b-468b-87a2-9216172ef4b1

## How I Tested These Changes

View launchpad, run page, global asset graph. Drag vertical and horizontal split panels, verify correct behavior. Verify that second-panel minimums are respected.

Test split panel contain

### CODE REVIEW

**Comment by hellendag** [comment]
* **#20909** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20909?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hellendag and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx [hunk 1]
```diff
@@ -13,220 +11,185 @@ interface SplitPanelContainerProps {
...
+      }
+      const storedSize = window.localStorage.getItem(key);
+      const parsed = storedSize === null ? null : parseFloat(storedSize);
+      return parsed === null || isNaN(parsed) ? firstInitialPercent : parsed;
+    }, [firstInitialPercent, key, second]);
+
+    const onChangeSize = useCallback(
+      (newValue: number) => {
+        window.localStorage.setItem(key, `${newValue}`);
+        setTrigger((current) => (current ? 0 : 1));
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx:51]: We're already tracking the `size` value in localStorage, so we don't need to duplicate that in local state.

File: js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx [hunk 2]
```diff
@@ -13,220 +11,185 @@ interface SplitPanelContainerProps {
...
-      firstPaneStyles.width = `calc(${firstSize}% - ${DIVIDER_THICKNESS}px)`;
+      firstPaneStyles.width = `calc(${firstSize / 100} * (100% - ${
+        DIVIDER_THICKNESS + (second ? secondMinSize : 0)
+      }px))`;
     } else {
       firstPaneStyles.minHeight = firstMinSize;
-      firstPaneStyles.height = `calc(${firstSize}% - ${DIVIDER_THICKNESS}px)`;
+      firstPaneStyles.height = `calc(${firstSize / 100} * (100% - ${
+        DIVIDER_THICKNESS + (second ? secondMinSize : 0)
+      }px))`;
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx:73]: The noteworthy calculation change to accommodate minimum size on the second panel.

File: js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx [hunk 3]
```diff
@@ -13,220 +11,185 @@ interface SplitPanelContainerProps {
...
 
       const firstPanelPercent =
-        this.props.axis === 'horizontal'
-          ? ((event.clientX - parentRect.left) * 100) / parentRect.width
-          : ((event.clientY - parentRect.top) * 100) / parentRect.height;
+        axis === 'horizontal'
+          ? ((event.clientX - parentRect.left) * 100) /
+            (parentRect.width - secondMinSize - DIVIDER_THICKNESS)
+          : ((event.clientY - parentRect.top) * 100) /
+            (parentRect.height - secondMinSize - DIVIDER_THICKNESS);
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx:125]: Ensures that the divider remains in the correct position with respect to the minimums applied.

**Review by salazarm — APPROVED** [review]
